src/python/visclaw/plotpages.py

```diff
@@ -549,7 +545,242 @@ def plots2latex(plot_pages_data):
+    htmlfile = plotdata._htmlfile
+    frametimef = plotdata._frametimef
+    allfigsfile = plotdata._allfigsfile
+    allframesfile = plotdata._allframesfile
+
+    numframes = len(framenos)
+    numfigs = len(fignos)
+
+    creationtime = current_time()
+
+    doc = KML.kml(
+        KML.Document())
+
+    for figname in plotdata._fignames:
+        plotfigure = plotdata.plotfigure_dict[figname]
+        figno = plotfigure.figno
+
+        if not figno in fignos:
+            continue
+
+        if not plotfigure.use_for_kml:
+            continue
+
+        ul = np.array([plotfigure.kml_xlimits[0], plotfigure.kml_ylimits[1]])
+        ur = np.array([plotfigure.kml_xlimits[1], plotfigure.kml_ylimits[1]])
+        lr = np.array([plotfigure.kml_xlimits[1], plotfigure.kml_ylimits[0]])
+
+        doc.Document.append(
+            KML.LookAt(KML.TimeSpan(),
+                KML.longitude((ul[0]+ur[0])/2),
+                KML.latitude((ur[1]+lr[1])/2),
+                KML.range(15000000),
+                KML.altitude(0),
+                KML.altitudeMode("absolute")))
+
+
+        # Open zip file
+        fname_kmz = plotdata.kml_index_fname + str(figno)
+        zip = zipfile.ZipFile("%s.kmz" % (fname_kmz),'w')
+
+        # Start doc.kml file (which will be head node in .kmz file)
+        fname_kml = "doc.kml"
+        print '\nCreating kml file %s for figure %d\n' % (fname_kml,figno)
+
+        filekml = open(fname_kml,'w')
+
+        filekml.write('<?xml version="1.0" encoding="UTF-8"?>\n')
+
+        # TODO : Make user options :
+        # (look at how dpi, use_for_kml, etc are set in setplot.py
+        #      -- gbegin  (time/date to start simulation)
+        #      -- range (height, in meters from sealevel)
+        #      -- tilt ?
+        #      -- Include "LookAt" tags ...
+        #      -- Add option to just print kml file with raw .png files?
+        #      -- Get rid of reliance on lxml/kml?
+        #      -- add URL options so users could get files from a server? (in this case,
+        #         we don't include .png files into zipped images)
+        #      -- Add color bar
+        #      -- Fix time slider so it starts at time zero (and not all files are loaded)
+        #      -- Add placemarks for Gauges files
+        #      -- Create index.html file as part of .kmz file
+        #      -- Different publishing options (.kmz will all files; .kmz with http links;
+        #         index.html file for Google Earth plug-in.)
+
+        try:
+            # If kml_starttime is already time_struct, we won't be able to
+            # extend it this try block will fail
+            user_time = plotfigure.kml_starttime      # 6 entries of a 9-tuple
+            user_time.extend([0,0,-1])                # Extend to 9 tuple
+        except:
+            user_time = time.gmtime()                 # 9-tuple local time
+
+        # This doesnt' really do the right thing.
+        starttime = time.mktime(user_time)
```

Comment:
I’ve actually fixed the time issue - will push that shortly. 
On Mar 20, 2015, at 2:16 PM, Kyle Mandli <notifications@github.com> wrote:

> In src/python/visclaw/plotpages.py:
> 
> > +        #      -- Fix time slider so it starts at time zero (and not all files are loaded)
> > +        #      -- Add placemarks for Gauges files
> > +        #      -- Create index.html file as part of .kmz file
> > +        #      -- Different publishing options (.kmz will all files; .kmz with http links;
> > +        #         index.html file for Google Earth plug-in.)
> > +
> > +        try:
> > +            # If kml_starttime is already time_struct, we won't be able to
> > +            # extend it this try block will fail
> > +            user_time = plotfigure.kml_starttime      # 6 entries of a 9-tuple
> > +            user_time.extend([0,0,-1])                # Extend to 9 tuple
> > +        except:
> > +            user_time = time.gmtime()                 # 9-tuple local time
> > +
> > +        # This doesnt' really do the right thing.
> > +        starttime = time.mktime(user_time)
> Would datetime work better for you here? I think it might do what you want.
> 
> —
> Reply to this email directly or view it on GitHub.
>